qt_gui/src/qt_gui/main.py

```diff
@@ -193,18 +193,20 @@ def _add_reload_paths(self, reload_importer):
 
     def _check_icon_theme_compliance(self):
         from python_qt_binding.QtGui import QIcon
-        # TODO find a better way to verify Theme standard compliance
-        if QIcon.fromTheme('document-save').isNull() or \
-           QIcon.fromTheme('document-open').isNull() or \
-           QIcon.fromTheme('edit-cut').isNull() or \
-           QIcon.fromTheme('object-flip-horizontal').isNull():
-            if platform.system() == 'Darwin' and \
-                    '/usr/local/share/icons' not in QIcon.themeSearchPaths():
-                QIcon.setThemeSearchPaths(QIcon.themeSearchPaths() + ['/usr/local/share/icons'])
-            original_theme = QIcon.themeName()
-            QIcon.setThemeName('Tango')
-            if QIcon.fromTheme('document-save').isNull():
-                QIcon.setThemeName(original_theme)
+        try:
+            # if themeName is defined we are on Linux, otherwise try to find the package
+            # qt_gui_icons
+            if(QIcon.themeName() == ''):
+                icon_paths = QIcon.themeSearchPaths()
+                _, package_path = get_resource('packages', 'qt_gui_icons')
```

Comment:
Since the content is never being used this should call `has_resource` instead.

I think this makes the other comment below obsolete.

Reply:
fixed https://github.com/ros-visualization/qt_gui_core/pull/222/commits/50d2dcc7e872eb482fb37cb7ac4098c0e86f4115

Reply:
The code is still calling `get_resource()` - and not as suggested `has_resource` instead?

Reply:
(Also in the `else` case `package_path` would not be a define variable but is being used in the next line.)

Reply:
I don't know why the interface is not updated. But I think i included your suggestion:


https://github.com/ros-visualization/qt_gui_core/blob/50d2dcc7e872eb482fb37cb7ac4098c0e86f4115/qt_gui/src/qt_gui/main.py#L200-L201

Reply:
There is no need to call `get_resource`. `has_resource` already returns the package path if the resource is found.